ledger/src/blockstore_processor.rs

```diff
@@ -714,6 +714,10 @@ pub struct ProcessOptions {
     /// This is useful for debugging.
     pub run_final_accounts_hash_calc: bool,
     pub use_snapshot_archives_at_startup: UseSnapshotArchivesAtStartup,
+    // Normally when we process blocks in blockstore we start from a snapshot where the bankforks only holds
+    // one bank. But during wen_restart we may need to replay the new blocks repaired, and the bankforks may
+    // hold any number of frozen banks.
```

Comment:
Isn't the find heaviest bank logic only run a single time after all the forks are already repaired, and we pass in an empty bank forks?

Reply:
The bank forks is reconstructed before entering wait_for_wenrestart, actually it's the result of [process_blockstore_from_root()](https://github.com/anza-xyz/agave/blob/master/core/src/validator.rs#L1256) upon restart, so at this point the bank forks should be populated with banks already on the validator.